src/bp/nlu-server/api.ts

```diff
@@ -0,0 +1,237 @@
+    trainSessionService = new TrainSessionService()
+    trainService = new TrainService(logger, engine, modelService, trainSessionService)
+  } catch (err) {
+    logger.attachError(err).error('an error occured while initializing the server')
+    process.exit(1)
+  }
+
+  app.get('/info', (req, res) => {
+    res.send({
+      version: options.version
+    })
+  })
+
+  const router = express.Router({ mergeParams: true })
+  router.post('/train', async (req, res) => {
+    try {
+      const input: TrainInput = await validate(req.body, TrainInputCreateSchema, {
+        stripUnknown: true
+      })
+
+      const { topics, entities, language, password } = input
+      const intents = _.flatMap(Object.values(topics))
+      const modelHash = engine.computeModelHash(intents, input.entities, input.language)
+
+      const seed = input.seed ?? Math.round(Math.random() * 10000)
+      const modelId = modelService.makeModelId(modelHash, input.language, seed)
+      const modelFileName = modelService.makeFileName(modelId, password)
+
+      // return the modelId as fast as possible
+      // tslint:disable-next-line: no-floating-promises
+      trainService.train(modelFileName, intents, entities, language, seed)
+
+      return res.send({
+        success: true,
+        modelId
+      })
+    } catch (err) {
+      res.status(500).send({
+        success: false,
+        error: err.message
+      })
+    }
+  })
+
+  router.get('/train/:modelId', async (req, res) => {
+    try {
+      const { modelId } = req.params
+      const { password } = req.query
+      const modelFileName = modelService.makeFileName(modelId, password ?? '')
+      let session = trainSessionService.getTrainingSession(modelFileName)
+      if (!session) {
+        const model = await modelService.getModel(modelFileName)
+
+        if (!model) {
+          return res.status(404).send({
+            success: false,
+            error: `no model or training could be found for modelId: ${modelId}`
+          })
+        }
+
+        session = {
+          key: modelId,
+          status: 'done',
+          progress: 1,
+          language: model!.languageCode
+        }
+      }
+
+      res.send({
+        success: true,
+        session
+      })
+    } catch (err) {
+      res.status(500).send({
+        success: false,
+        error: err.message
+      })
+    }
+  })
+
+  router.post('/train/:modelId/cancel', async (req, res) => {
+    try {
+      const { modelId } = req.params
+      const { password } = req.body
+      const modelFileName = modelService.makeFileName(modelId, password ?? '')
+      const session = trainSessionService.getTrainingSession(modelFileName)
+      if (session?.status === 'training') {
+        await engine.cancelTraining(modelFileName)
+        return res.send({
+          success: true
+        })
+      }
+      res.status(404).send({
+        success: true,
+        error: `no current training for model id: ${modelId}`
+      })
```

Comment:
Please add new lines to make the code clearer, for ex you could split your const block from your if, then another before your res.status